### PR TITLE
feat: add oraclecloud.com platform support

### PR DESCRIPTION
**Pull request by sergelogvinov**
Add Oracle Cloud platform.

* cloud-init for oraclecloud (IMDSv2)
* amd64/arm64 arch support
* set DHCPv6 on if IPv6 subnet allocated like in Azure

## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [x] you formatted your code (`make fmt`)
- [x] you linted your code (`make lint`)
- [x] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

---
This change is [Reviewable](https://reviewable.io/reviews/talos-systems/talos/4712)

### CODE REVIEW

File: website/content/docs/v0.14/Cloud Platforms/oracle.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: "Oracle"
```
Comment (smira) [inline website/content/docs/v0.14/Cloud Platforms/oracle.md:2]: this should go under v0.15 once we cut them (I will try to do this today)

**Review by smira — COMMENTED** [review]
feels like `Makefile` changes are missing for the new image type

and also in `.drone.jsonnet` for the new assets to be part of the release

File: website/content/docs/v0.14/Cloud Platforms/oracle.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
```
Comment (frezbo) [inline website/content/docs/v0.14/Cloud Platforms/oracle.md:1]: is there a little draft docs available? Would be great to have some steps, even if it's a draft or manual process

File: website/content/docs/v0.14/Cloud Platforms/oracle.md (hunk 1 above)
Comment (sergelogvinov) [reply website/content/docs/v0.14/Cloud Platforms/oracle.md]: Yep, definitely

**Comment by kurious-diru** [comment]
Hey @sergelogvinov !

I tested this setup in oracle cloud, it works fine if we give the config data while creating an instance, but it isn't picking up while running in maintanence mode, I think the public IP of the VM isn't exposed for talosctl to apply the configuration.

**Comment by smira** [comment]
> Hey @sergelogvinov !
> 
> I tested this setup in oracle cloud, it works fine if we give the config data while creating an instance, but it isn't picking up while running in maintanence mode, I think the public IP of the VM isn't exposed for talosctl to apply the configuration.
> 

This is known deficiency in the way Talos works on cloud platform, I hope to get it fixed next week

**Review by smira — APPROVED** [review]

**Review by smira — CHANGES_REQUESTED** [review]
I think we need to clean up docs a bit before we can merge it

**Review by smira — APPROVED** [review]

**Pull request merged**